src/fileformats/ocd_file_import.cpp

```diff
@@ -1771,6 +1833,188 @@ void OcdFileImport::setupPointSymbolPattern(PointSymbol* symbol, std::size_t dat
 	}
 }
 
+
+QImage OcdFileImport::importSymbolIcon(const Ocd::IconV8& icon)
+{
+	static const QColor palette[16] = {
+	    {   0,   0,   0 },
+	    { 128,   0,   0 },
+	    { 0,   128,   0 },
+	    { 128, 128,   0 },
+	    {   0,   0, 128 },
+	    { 128,   0, 128 },
+	    {   0, 128, 128 },
+	    { 128, 128, 128 },
+	    { 192, 192, 192 },
+	    { 255,   0,   0 },
+	    {   0, 255,   0 },
+	    { 255, 255,   0 },
+	    {   0,   0, 255 },
+	    { 255,   0, 255 },
+	    {   0, 255, 255 },
+	    { 255, 255, 255 }
+	};
+	
+	auto icon_bits = icon.bits;
+	constexpr int icon_size = 22;
+	QImage image(icon_size, icon_size, QImage::Format_ARGB32_Premultiplied);
+	for (int y = icon_size - 1; y >= 0; --y)
+	{
+		for (int x = 0; x < icon_size; x += 2)
+		{
+			image.setPixelColor(x, y, palette[(*icon_bits) >> 4]);
+			image.setPixelColor(x+1, y, palette[*(icon_bits++) & 0xf]);
+		}
+		icon_bits++;
+	}
+	return image;
```

Comment:
Just dreaming....
```
QImage image(icon.bits, icon_size, icon_size, QImage::Format_Indexed8);
image.setColorTable(palette);
return mirrored(true, false).convertToFormat(QImage::Format_ARGB32_Premultiplied);
```
would be my starting point for implementation of the above. It of course assumes that palette is a `QVector`.

Reply:
Hmm, this of course applies to v9 icon format. There is nothing like `QImage::Format_Indexed4` which could be leveraged here.

Reply:
I leave this as an exercise to be done.